src/arch/x86/analyzex86.rs

```diff
@@ -454,27 +423,34 @@ pub struct MemoryManager<'a>
     pub list: Vec<MemoryBounds<'a>>,
 }
 
-fn read_int<T: Num + BitOr<Output=T> + NumCast + Copy>(
-     _byte: &mut T,
+pub fn read_int<T: Num + NumCast + Copy + 'static>(
      _offset: usize,
      _address: usize,
-     _size: usize,
-     _binary: &[u8])-> bool
+     _binary: &[u8]) -> Option<T>
+    where u64: AsPrimitive<T>
 {
-    let length = ::std::mem::size_of::<T>();
+    if let Some(start) = _offset.checked_sub(_address) {
+        // start does not underflow, good!
 
-    for offset in 0..length {
-        let mut _byte0: u8 = 0;
-        let address_offset = _offset-_address+offset;
-        
-        if address_offset >= _size{
-            return false;
+        let count_to_read = ::std::mem::size_of::<T>();
+
+        if let Some(end) = start.checked_add(count_to_read) {
+            if end <= _binary.len() {
+                // end is in bounds, also good!
+
+                let mut result: u64 = 0;
+
+                for byte_index in 0..count_to_read {
+                    result |= (_binary[start + byte_index] as u64) << (byte_index * 8);
+                }
+
+                return NumCast::from::<u64>(result)
+                    .map(|x: u64| x.as_());
```

Comment:
This plus the additional type bounds is a bit of trickery to cast `result` to `T` the same as would have been done with `as`, which lets this function generally read signed or unsigned values without issue. Which means the `get_int` and `get_int_signed` methods later can mostly just use this! [relevant docs](https://docs.rs/num/0.2.0/num/cast/trait.AsPrimitive.html)